src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -717,48 +721,61 @@ public async Task Delete()
                     auditLog.Info($"Delete [{string.Join(", ", parameters.DatabaseNames)}] database from ({string.Join(", ", parameters.FromNodes)}) by {clientCert?.Subject} ({clientCert?.Thumbprint})");
                 }
 
-                if (parameters.FromNodes != null && parameters.FromNodes.Length > 0)
+                using (context.OpenReadTransaction())
                 {
-                    using (context.OpenReadTransaction())
+                    var fromNodes = parameters.FromNodes != null && parameters.FromNodes.Length > 0;
+
+                    foreach (var databaseName in parameters.DatabaseNames)
                     {
-                        foreach (var databaseName in parameters.DatabaseNames)
+                        DatabaseTopology topology = null;
+                        using (var rawRecord = ServerStore.Cluster.ReadRawDatabaseRecord(context, databaseName))
                         {
-                            DatabaseTopology topology;
-                            using (var rawRecord = ServerStore.Cluster.ReadRawDatabaseRecord(context, databaseName))
+                            if (rawRecord == null)
+                                continue;
+
+                            switch (rawRecord.LockMode)
                             {
-                                if (rawRecord == null)
+                                case DatabaseLockMode.Unlock:
+                                    databasesToDelete.Add(databaseName);
+                                    break;
+                                case DatabaseLockMode.PreventDeletesIgnore:
```

Comment:
Add a note to the log? Maybe an alert?
For that matter, if this is set, we probably want to record who is trying the delete (ip, cert, etc, no)?

Reply:
We are recording that beforehand (auditlog). Maybe we actually should record that after (or both?) ?

Reply:
Audit log may not be configured.